lib/static/components/controls/browser-list/index.jsx

```diff
@@ -5,224 +5,172 @@ import {flatten, isEmpty, get, chain, compact} from 'lodash';
+            })
         });
-
-        return flatten(selectedItems);
+        return selectedOptions;
     }
-
-    function prepareTreeData() {
-        const mkNode = ({browserId, version, icon}) => ({
-            icon,
-            data: {browserId, version},
-            value: buildComplexId(browserId, version),
-            label: mkLabel({browserId, version, elements})
-        });
-
-        return available.map(({id: browserId, versions}) => {
-            const node = mkNode({browserId, icon: mkBrowserIcon(browserId)});
-
-            if (!isEmpty(versions) && versions.length > 1) {
-                node.children = versions.map(version => mkNode({browserId, version}));
+    const rawOptions =  useMemo(getOptions, [available]);
+    const getOptionsFrom = (optionsData) => {
+        const allOptionsList = [];
+        optionsData.forEach((option) => {
+            if (option.label) {
+                getOptionsFrom(option.options).forEach((o) => allOptionsList.push(o));
             }
-
-            return node;
-        });
+            else {
+                allOptionsList.push(option.value);
+            }
+        })
+        return allOptionsList;
+    }
+    const allOptions =  useMemo(() => getOptionsFrom(rawOptions), [rawOptions]);
+    const options = useSelectOptions({
+        options: rawOptions
+    });
+    const mapping = useMemo(getMapping, [available]);
+    const [selected, setSelected] = useState(getSelected());
+
+    const selectAll = () => {
+        setSelected(allOptions);
     }
 
-    function _selectAll() {
-        const leaves = Object.keys(treeDataMap).filter(key => treeDataMap[key].isLeaf);
-        const cb = selected => selected.length !== leaves.length
-            ? leaves
-            : selected;
+    const deselectAll = () => {
+        setSelected([]);
+    }
 
-        setSelected(cb);
+    const formatSelectedData = () => {
+        const selectedData = {}
+        selected.forEach((option) => {
+            if (!mapping[option] || !mapping[option].id) return;
+            const {id: browserId, version} = mapping[option];
+            selectedData[browserId] = selectedData[browserId] || [];
+            selectedData[browserId].push(version);
+        })
+        return Object.entries(selectedData).map(([id, versions]) => ({id, versions: compact(versions)}))
     }
 
-    function mkLabel({browserId, version, elements}) {
+    const renderFilter = () => {
+        const allSelected = selected.length == options.length;
         return (
-            <Label
-                treeDataMap={treeDataMap}
-                browserId={browserId}
-                version={version}
-                elements={elements}
-                setSelected={setSelected}
-            />
-        );
+            <div className='browserlist__filter'>
+            <Button onClick={allSelected ? deselectAll : selectAll} width='max'>
+                {allSelected ? 'Deselect All' : 'Select All'}
+            </Button>
```

Comment:
Does it make sense? As far as i know, when all of the browsers are deselected, html-reporter would show all of the browsers. So user would be confused: "why are there browsers shown, if nothing is selected? Why did nothing change when i pressed 'deselect'"

I think, as long as user has "Only" button, "Deselect all" only makes it worse